idaes/core/base/property_meta.py

```diff
@@ -104,60 +104,79 @@ class UnitSet(object):
     Units of measurement are defined by setting units for the seven base SI quantities
     (amount, current, length, luminous intensity, mass, temperature and time), from which units
     for all other quantities are derived. The units of the seven base quantities must be provided
-    when instantiating the UnitSet - units are optional for current and luminous intensity.
+    when instantiating the UnitSet, otherwise base SI units are assumed.
 
     Units can be accesses by via either a property on the UnitSet (e.g., UnitSet.TIME) or
     via an index on the UnitSet (e.g., UnitSet["time"]).
     """
 
-    # TODO: Test this in isolation
-    def __init__(self, **kwds):
-        self._time = kwds.pop("time", None)
-        self._length = kwds.pop("length", None)
-        self._mass = kwds.pop("mass", None)
-        self._amount = kwds.pop("amount", None)
-        self._temperature = kwds.pop("temperature", None)
-        self._current = kwds.pop("current", None)
-        self._luminous_intensity = kwds.pop("luminous_intensity", None)
+    base_quantities = [
+        "time",
+        "length",
+        "mass",
+        "amount",
+        "temperature",
+        "current",
+        "luminous_intensity",
+    ]
+
+    def __init__(
+        self,
+        amount: _PyomoUnit = units.mol,
+        current: _PyomoUnit = units.watt,
+        length: _PyomoUnit = units.meter,
+        luminous_intensity: _PyomoUnit = units.candela,
+        mass: _PyomoUnit = units.kilogram,
+        temperature: _PyomoUnit = units.kelvin,
+        time: _PyomoUnit = units.seconds,
+        **kwds,
```

Comment:
If you are going to put all the base units explicitly, then can we drop `**kwds` (at least until you are going to start supporting user-specified derived units)?

Reply:
I was debating that - I wondered if it would be useful to have a more useful error message for extra arguments, but there does come a point where you can't fix stupid.